### ISSUE TITLE
Infinite loop and memory crash with $$(....)[xx].$(...) when the element is not found

### ISSUE THREAD

**Issue by lacell75** [webdriverio/webdriverio#6798]
*Environment (please complete the following information):**
 - **WebdriverIO version:** 7.5.2
 - **Mode:** WDIO
 - **If WDIO Testrunner, running sync/async:** sync
 - **Node.js version:** 14.16
 - **NPM version:** 6.14.11
 - **Browser name and version:** Any
 - **Platform name and version:**  Windows 10
 - **Additional wdio packages used (if applicable):** "@wdio/cli": "^7.5.2",
    "@wdio/cucumber-framework": "^7.5.2",
    "@wdio/dot-reporter": "^7.5.2",
    "@wdio/local-runner": "^7.5.2",
    "@wdio/mocha-framework": "^7.5.2",
    "@wdio/spec-reporter": "^7.5.2",
    "@wdio/sync": "^7.5.2",
    "@wdio/allure-reporter": "^7.5.2",
     "expect": "^26.6.2",
    "expect-webdriverio": "^2.0.1",

**Describe the bug**
When I try to select an element in a list of elements, if the element is not found and its position is not at the beginning of the list. I have a infinite loop and a memory crash

**To Reproduce**
https://webdriver.io/docs/api/browser/$$

```
it('should return false', () => {
  browser.url('https://webdriver.io/docs/api');
  const list = $$('.menu__list-item');
  console.log(list.length); // 107
  console.log(list[50].$('[icon="me"]').isDisplayed()); // Expected false
});
```

if I select the second element of the list, that works:
```
it('should return false', () => {
  browser.url('https://webdriver.io/docs/api');
  const list = $$('.menu__list-item');
  console.log(list.length); // 107
  console.log(list[2].$('[icon="me"]').isDisplayed()); // false is displayed
});
```

**Expected behavior**
The result should be displayed 

**wdio.config file:**
```
const expect = require('expect');
require('expect-webdriverio').setOptions({ wait: 5000 });
waitforTimeout: 5000,
 waitforInterval: 5000,
  //
  // Default timeout in milliseconds for request
  // if Selenium Grid doesn't send response
  connectionRetryTimeout: 240000,
  //
  // Default request retries count
  connectionRetryCount: 0,

 framework: 'cucumber',

  reporters: ['spec',
    ['allure', {
      outputDir: 'report/allure-results',
      disableWebdriverStepsReporting: false,
      disableWebdriverScreenshotsReporting: false,
      useCucumberStepReporter: true,
    }]],
  cucumberOpts: {
    // <boolean> show full backtrace for errors
    backtrace: false,
    // <string[]> module used for processing required features
    requireModule: ['@babel/register'],
    // <boolean< Treat ambiguous definitions as errors
    failAmbiguousDefinitions: true,
    // <boolean> invoke formatters without executing steps
    // dryRun: false,
    // <boolean> abort the run on first failure
    failFast: false,
    // <boolean> Enable this config to treat undefined definitions as
    // warnings
    ignoreUndefinedDefinitions: false,
    // <string[]> ("extension:module") require files with the given
    // EXTENSION after requiring MODULE (repeatable)
    name: [],
    // <boolean> hide step definition snippets for pending steps
    snippets: true,
    // <boolean> hide source uris
    source: true,
    // <string[]> (name) specify the profile to use
    profile: [],
    // <string[]> (file/dir) require files before executing features
    require: [
      './src/steps/given.js',
      './src/steps/then.js',
      './src/steps/when.js',
      './src/steps/after.js',
      // Or search a (sub)folder for JS files with a wildcard
      // works since version 1.1 of the wdio-cucumber-framework
      // './src/**/*.js',
    ],
    // <string> specify a custom snippet syntax
    snippetSyntax: undefined,
    // <boolean> fail if there are any undefined or pending steps
    strict: true,
    // <string> (expression) only execute the features or scenarios with
    // tags matching the expression, see
    // https://docs.cucumber.io/tag-expressions/
    tagExpression: 'not @Pending',
    // <boolean> add cucumber tags to feature or scenario name
    tagsInTitle: true,
    // <number> timeout for step definitions
    timeout: 99999999,
  },
```

**Comment by lacell75** [comment]
After investigation, I think the issue is in [`middlewares`](https://github.com/webdriverio/webdriverio/blob/main/packages/webdriverio/src/middlewares.ts) or [`getElementObject`](https://github.com/webdriverio/webdriverio/blob/main/packages/webdriverio/src/utils/getElementObject.ts)
If we select the first element of elements array $$(...)[0], the elementErrorHandler is called once. But if we want to get the nth element of the array , the function is called n times(why?). If the array is big and the element doesn't exist, we have many ping pong between elementErrorHandler and refetchElement functions and the error is never throw or recovered.

I need help from an expert to fix it.

**Comment by christian-bromann** [comment]
Fixed by #6848
